### PR TITLE
[REEF-206]  Create NuGet for Org.Apache.REEF.Bridge.JAR

### PR DESCRIPTION
**Pull request by jwang98052**
This PR is to trigger the NuGet creation for Org.Apache.REEF.Bridge.JAR.
Added NuGet spec for REEF.Bridge.Jar
updated NuGet.Target
Updated csproj files for NuGet creation for REEF.Bridge.Jar and REEF.All
Updated platform to match  x64

JIRA: REEF-206. (https://issues.apache.org/jira/browse/REEF-206)

Author: Julia Wang  Email: jwang98052@yahoo.com

### CODE REVIEW

File: lang/cs/.nuget/NuGet.targets [hunk 1]
```diff
@@ -87,7 +87,7 @@ under the License.
 
         <!-- Commands -->
         <RestoreCommand>$(NuGetCommand) install "$(PackagesConfig)" -source "$(PackageSources)"  $(NonInteractiveSwitch) $(RequireConsentSwitch) -solutionDir $(PaddedSolutionDir)</RestoreCommand>
-        <BuildCommand>$(NuGetCommand) pack "$(FinalizedNuspecFile)" -BasePath $(NugetProjectPath) -Properties "Configuration=$(Configuration);Platform=$(Platform)" $(NonInteractiveSwitch) -OutputDirectory "$(PackageOutputDir)" </BuildCommand>
+        <!--BuildCommand>$(NuGetCommand) pack "$(FinalizedNuspecFile)" -BasePath $(NugetProjectPath) -Properties "Configuration=$(Configuration);Platform=$(Platform);REEF_Version=$(REEF_Version)" $(NonInteractiveSwitch) -OutputDirectory "$(PackageOutputDir)" </BuildCommand -->
```
Comment (markusweimer) [inline lang/cs/.nuget/NuGet.targets]: Please remove the commented out code.

File: lang/cs/Org.Apache.REEF.Bridge.JAR/Org.Apache.REEF.Bridge.JAR.csproj [hunk 2]
```diff
@@ -16,14 +16,20 @@ specific language governing permissions and limitations
 under the License.
 -->
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="12.0">
+    <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   <PropertyGroup>
     <!--Set configuration parameters usually set in the Solution. This is needed for build.props to work.-->
-    <SolutionDir Condition="'$(SolutionDir)' == ''">$([System.IO.Path]::GetFullPath($(MSBuildProjectDirectory)\..))</SolutionDir>
+    <!--<SolutionDir Condition="'$(SolutionDir)' == ''">$([System.IO.Path]::GetFullPath($(MSBuildProjectDirectory)\..))</SolutionDir> -->
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Bridge.JAR/Org.Apache.REEF.Bridge.JAR.csproj]: Please remove the commented out code.

File: lang/cs/Org.Apache.REEF.Wake/Org.Apache.REEF.Wake.csproj [hunk 3]
```diff
@@ -19,7 +19,7 @@ under the License.
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <Platform Condition=" '$(Platform)' == '' ">x64</Platform>
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Wake/Org.Apache.REEF.Wake.csproj]: We have this line in every csproj file. Would it make sense to move it to build.props?

**Comment by markusweimer** [comment]
When I build using `msbuild Org.Apache.REEF.sln /p:Configuration="Release" /p:Platform="x64"`, I get the following warnings:

```
"c:\src\reef\lang\cs\Org.Apache.REEF.sln" (default target) (1) ->
"c:\src\reef\lang\cs\Org.Apache.REEF.Client\Org.Apache.REEF.Client.csproj" (default target) (15) ->
(ResolveAssemblyReferences target) ->
  C:\Program Files (x86)\MSBuild\12.0\bin\Microsoft.Common.CurrentVersion.targets(1697,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the ref erence "c:\src\reef\lang\cs\bin\x64\Release\Org.Apache.REEF.Bridge\Org.Apache.REEF.Bridge.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Con figuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [c:\src\r eef\lang\cs\Org.Apache.REEF.Client\Org.Apache.REEF.Client.csproj]


"c:\src\reef\lang\cs\Org.Apache.REEF.sln" (default target) (1) ->
"c:\src\reef\lang\cs\Org.Apache.REEF.Tests\Org.Apache.REEF.Tests.csproj" (default target) (18) ->
  C:\Program Files (x86)\MSBuild\12.0\bin\Microsoft.Common.CurrentVersion.targets(1697,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the ref erence "c:\src\reef\lang\cs\bin\x64\Release\Org.Apache.REEF.Bridge\Org.Apache.REEF.Bridge.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Con figuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [c:\src\r eef\lang\cs\Org.Apache.REEF.Tests\Org.Apache.REEF.Tests.csproj]


"c:\src\reef\lang\cs\Org.Apache.REEF.sln" (default target) (1) ->
"c:\src\reef\lang\cs\Org.Apache.REEF.All\Org.Apache.REEF.All.csproj" (default target) (19) ->
(BuildPackage target) ->
  EXEC : warning : 3 issue(s) found with package 'Org.Apache.REEF.All'. [c:\src\reef\lang\cs\Org.Apache.REEF.All\Org.Apache.REEF.All.csproj]
```

Are those expected?

**Comment by markusweimer** [comment]
Also, when I open the project in VS, I get the following warnings:

```
Warning	1	There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\src\reef\lang\cs\bin\x64\Release\Org.Apache.REEF.Bridge\Org.Apache.REEF.Bridge.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.	Org.Apache.REEF.Client
```

And:

```
Warning	2	There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "C:\src\reef\lang\cs\bin\x64\Release\Org.Apache.REEF.Bridge\Org.Apache.REEF.Bridge.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.	Org.Apache.REEF.Tests
```

**Comment by markusweimer** [comment]
The NuGet contains the jar in a folder called `jars`. They are not copied to downstream project's output folders. I assume this is done in another PR?

**Comment by jwang98052** [comment]
It is known issue. I have synced everywhere to use x64. But warning still comes out. Jira #219 is created to follow up it.

**Pull request closed without merging**